driver/bpf/fillers.h

```diff
@@ -249,13 +249,24 @@ FILLER(sys_open_x, true)
 	if (res != PPM_SUCCESS)
 		return res;
 
-	/*
-	 * Device
-	 */
 	if (retval < 0 || !bpf_get_fd_dev_ino(retval, &dev, &ino))
```

Comment:
Looking at the new code, I would propose you a final change. What do you think if we initialize at the beginning of the function `dev` and `ino` to `0` and then we don't care about what `bpf_get_fd_dev_ino` does internally... If we are able to take the right values OK, otherwise we will simply push 0 for `dev` and `ino` since we have already initialized them. We could also move the check `retval < 0` inside `bpf_get_fd_dev_ino` like in the kernel module function `get_fd_dev_ino`, in this way we are no more interested in returning something from `bpf_get_fd_dev_ino` and we can clean a little bit the code.

So from this code:

```c
int dev;
int ino;

// ...

if (retval < 0 || !bpf_get_fd_dev_ino(retval, &dev, &ino))
{
    dev = 0;
    ino = 0;
}


/*
 * Device
 */
res = bpf_val_to_ring(data, dev);
if (res != PPM_SUCCESS)
    return res;
```

We can obtain something like this:


```c
int dev = 0;
int ino = 0;

// ...

/* This returns void and checks the `retval` internally. */
bpf_get_fd_dev_ino(retval, &dev, &ino);

/*
 * Device
 */
res = bpf_val_to_ring(data, dev);
if (res != PPM_SUCCESS)
    return res;
```


In the same way, the kernel module does a similar thing in the `get_fd_dev_ino` function. It checks the `fd` inside itself but it returns a boolean while, also in this case, we could obtain a void function with the same initializiation phase we have done in BPF, so something like that

```c
int dev = 0;
int ino = 0;

// ...

get_fd_dev_ino(retval, &dev, &ino);

/*
 *  dev
 */
res = val_to_ring(args, dev, 0, false, 0);
if (unlikely(res != PPM_SUCCESS))
    return res;

```

This would also allow us to have a sort of "implementation parity" between kernel module and BPF probe. Let me know what do you thing about this approach, maybe there is something that I missed. Thank you again for you effort in doing this!

Reply:
I agree, this simplifies code. Moving `retval` check into `bpf_get_fd_dev_ino` also makes sense, we'd want to do it every time anyway (before or after calling the function). Just made some changes.